src/main/java/org/folio/oaipmh/dao/impl/SetDaoImpl.java

```diff
@@ -120,14 +118,28 @@ public SetDaoImpl(final PostgresClientFactory postgresClientFactory) {
       }));
   }
 
+  @Override
+  public Future<SetCollection> getSetList(int offset, int limit, String tenantId) {
+    return getQueryExecutor(tenantId).transaction(queryExecutor -> queryExecutor.query(dslContext -> dslContext.selectFrom(SET)
+      .offset(offset)
+      .limit(limit))
+      .map(this::queryResultToSetCollection));
+  }
+
+  private SetCollection queryResultToSetCollection(QueryResult queryResult) {
+    List<Set> list = queryResult.stream()
+      .map(row -> rowToSet(row.unwrap()))
+      .collect(Collectors.toList());
+    return new SetCollection().withSets(list);
+  }
+
   private void prepareSetMetadata(Set entry, String userId, InsertType insertType) {
-    if (insertType.equals(InsertType.UPDATE)) {
-      entry.setUpdatedDate(from(Instant.now()));
-      entry.setUpdatedByUserId(userId);
-    } else {
+    if (insertType.equals(InsertType.INSERT)) {
```

Comment:
Why do we need to manually manage metadata?  Can't the database/RMB do that for us?

Reply:
> Why do we need to manually manage metadata? Can't the database/RMB do that for us?

I don't think so, there are will be changes relatively to metadata soon, it will be leaded to the common metadata strucutre as it used on the project. We don't want to save it as nested json object to database (I mean the json field type) cause we have to implement seraching mechanism which uses metadata fields and this won't be optimize to parse this json within searching logic.

Reply:
> Why do we need to manually manage metadata? Can't the database/RMB do that for us?

As well we don't use rmb to manage our database records, we use another approach (liquibase + jooq) in compare with other folio modules.